app/src/main/java/com/islaharper/dawnofandroid/presentation/components/RatingBar.kt

```diff
@@ -0,0 +1,231 @@
+package com.islaharper.dawnofandroid.presentation.components
+
+import android.content.res.Configuration.UI_MODE_NIGHT_YES
+import android.util.Log
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.drawscope.clipPath
+import androidx.compose.ui.graphics.drawscope.scale
+import androidx.compose.ui.graphics.drawscope.translate
+import androidx.compose.ui.graphics.vector.PathParser
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import com.islaharper.dawnofandroid.domain.model.DawnOfAndroidTheme
+import com.islaharper.dawnofandroid.domain.model.Rating
+import com.islaharper.dawnofandroid.ui.theme.DawnOfAndroidTheme
+import com.islaharper.dawnofandroid.ui.theme.LightStarBackgroundColour
+import com.islaharper.dawnofandroid.ui.theme.LightStarColour
+import com.islaharper.dawnofandroid.ui.theme.PADDING_SMALL
+import com.islaharper.dawnofandroid.util.Constants.STAR_PATH
+
+@Composable
+fun RatingBar(
+    rating: Double,
+    modifier: Modifier = Modifier,
+    spaceBetween: Dp = PADDING_SMALL
+) {
+    val result = calculateRating(rating = rating)
```

Comment:
We don't want `calculateRating(rating = rating)` to be executed every time. I think it should be something like:

```kotlin
    val result by remember(rating) { mutableStateOf(calculateRating(rating = rating)) }
```